Update extrema bounding method for compute="eccentricities" parameter

yusuf-csdev opened the pull request:
Issue number: [#5402](https://github.com/networkx/networkx/issues/5402)

I have updated extrema_bounding function's documentation:
    - In parameters section, "eccentricities" is listed with explanation,
    - In returns section, eccentricities' return value is listed
    - In raises section, NetworkXError to be raised when compute parameter is passed an invalid argument is written
    
In extrema_bounding function, when compute parameter is passed  "eccentricities" argument, the ruled_out variable is set to an empyt set (rather than an empty dict). Moreover, if the compute parameter is not passed a valid extreme distance metric argument (not one of the 'diameter', 'radius', 'periphery', 'center', 'eccentricities'), a NetworkXError is raised with appropriate message.

rossbar commented on networkx/algorithms/distance_measures.py:
```diff
@@ -125,32 +128,31 @@ def extrema_bounding(G, compute="diameter"):
+            ruled_out = set()
+        else:
+            msg = "The argument passed to compute parameter is invalid. Please enter one of the following extreme distance metrics: 'diameter', 'radius', 'periphery', 'center', 'eccentricities'"
+            raise nx.NetworkXError(msg)
```
```suggestion
            raise ValueError(msg)
```
I think a `ValueError` would be more appropriate here.

rossbar commented on networkx/algorithms/distance_measures.py:
```diff
@@ -30,19 +30,22 @@ def extrema_bounding(G, compute="diameter"):
     NetworkXError
-        If the graph consists of multiple components
+        If the graph consists of multiple components or
+        If the compute parameter is passed an invalid argument
```
```suggestion
    ValueError
        If `compute` is not one of "diameter", "radius", "periphery",
        "center", or "eccentricities".
```
Minor suggestion to make this more informative. See below for suggestion on changing the exception type.

rossbar commented on networkx/algorithms/distance_measures.py:
```diff
@@ -125,32 +128,31 @@ def extrema_bounding(G, compute="diameter"):
-            ruled_out = {}
+            ruled_out = set()
+        else:
+            msg = "The argument passed to compute parameter is invalid. Please enter one of the following extreme distance metrics: 'diameter', 'radius', 'periphery', 'center', 'eccentricities'"
```
```suggestion
            msg = "compute must be one of 'diameter', 'radius', 'periphery', 'center', 'eccentricities'"
```
Just a suggestion for a shorter exception msg that conveys the important info.

rossbar reviewed (commented):
The docstring changes and exception look good to me, though I think the exception should be a `ValueError`.

It'd also be nice to add a test for the new exception (even though `extrema_bounding` is currently untested!). For example, maybe add something like:

```python
def test_extrema_bounding_invalid_compute_kwarg():
    G = nx.path_graph(3)
    with pytest.raises(ValueError, match="compute must be one of"):
        nx.extrema_bounding(G, compute="spam")
```

to `networkx/algorithms/tests/test_distance_measures.py`

yusuf-csdev commented:
Thank you for your comments. I have updated the documentation accordingly. In test document, I have created a new class called "TestExtremaBounding", and added the method you suggested.

rossbar reviewed (approved):
Thanks @yusuf-csdev , this LGTM!

I went ahead and removed the addition of the test class as that's an organization pattern NX is trying to avoid when it's not explicitly necessary.

The added test + docstring updates are an obvious improvement so I think this should go in independent of the discussion in #5415
